zerver/views/__init__.py

```diff
@@ -764,7 +773,7 @@ def create_realm(request, creation_key=None):
                               request=request)
 
 def confirmation_key(request):
-    return json_success(request.session.get('confirmation_key'))
+    return json_success({'confirmation_key': os.getenv('CONFIRMATION_KEY')})
```

Comment:
(Is the bug this is intended to fix still reproducible?  I can imagine this being something you saw before you set SESSION_COOKIE_DOMAIN)